pkg/cache/cache.go

```diff
@@ -360,6 +360,37 @@ func (c *Cache) ForgetWorkload(w *kueue.QueuedWorkload) error {
 	return nil
 }
 
+// UpdateWorkloadStatus is a wrapper to update the condition of a workload
+// sets a new current condition and stores it on Conditions
+func (c *Cache) UpdateWorkloadStatus(ctx context.Context, wl *kueue.QueuedWorkload, wlct kueue.QueuedWorkloadConditionType, message, reason string) error {
+	c.Lock()
+	defer c.Unlock()
```

Comment:
why do we need to lock? the cache is not being updated

Reply:
removed